core/src/main/java/apoc/load/Xml.java

```diff
@@ -69,6 +69,7 @@ public class Xml {
     private static final XMLInputFactory FACTORY = XMLInputFactory.newFactory();
     static {
         FACTORY.setProperty(XMLInputFactory.IS_COALESCING, true);
+        FACTORY.setProperty(XMLInputFactory.SUPPORT_DTD, false);
         FACTORY.setProperty(XMLInputFactory.IS_SUPPORTING_EXTERNAL_ENTITIES, false);
```

Comment:
not a Java expert, but what is the difference between this implementation and what OWASP snippet looks like: `xmlInputFactory.setProperty("javax.xml.stream.isSupportingExternalEntities", false);`

Reply:
also , what about this one:
`// This causes XMLStreamException to be thrown if external DTDs are accessed.
xmlInputFactory.setProperty(XMLConstants.ACCESS_EXTERNAL_DTD, "");`.
I guess it is not needed , because right above we set `SUPPORT_DTD` to false?

Reply:
>not a Java expert, but what is the difference between this implementation and what OWASP snippet looks like: xmlInputFactory.setProperty("javax.xml.stream.isSupportingExternalEntities", false);

`FACTORY.setProperty(XMLInputFactory.IS_SUPPORTING_EXTERNAL_ENTITIES, false);` is the same as the line above. It's just using the Enum object rather than writing it out as a String.

> also , what about this one:
> // This causes XMLStreamException to be thrown if external DTDs are accessed. 
> xmlInputFactory.setProperty(XMLConstants.ACCESS_EXTERNAL_DTD, "");.
> I guess it is not needed , because right above we set SUPPORT_DTD to false?

I think the OWASP article was for an older version of Java perhaps, where the StAX parser returned was using this configuration. The default parser Java is giving us does not use the setting. 

The setting is not entirely needed however as it simply throws an exception whenever the parser encounters a `<!DOCTYPE>`. Currently what our implementation is doing is that it simply ignores the `<!DOCTYPE>` and will throw an error if it encounters an unknown entity (eg one that was declared the in DOCTYPE)

This is not a bad thing really as it makes my change a bit more backwards compatible while still being safe.

It's also perfectly aligned with the LeftShift (eg competitor to Snyk and Veracode) recommendation which is to implement exactly what I've done. It's mostly aligned with the OWASP article but only because I suspect the OWASP article is quite old and doesn't fully work anymore.

Reply:
Thanks for looking into this!! Let's merge :)